### ISSUE TITLE
[Feature Request] Torrent Searching

### ISSUE THREAD

**Issue by zero77** [SchizoDuckie/DuckieTV#618]
Can you get DuckieTV to try more than one search provider if it doesn’t find the torrent in the first one.

**Comment by Js41637** [comment]
This is on the todo list, somewhere.

**Comment by zero77** [comment]
O really sorry. Where is the to do list located.

**Comment by Js41637** [comment]
Yeah I'm not sure if it's actually listed but it is something we've been meaning to add.

**Comment by zero77** [comment]
Is the to do list publicly listed somewhere and if not it would be great if you could add it.

**Comment by Js41637** [comment]
Its just the tickets here on github is what we use as a list to track things we're going to add, I dont think we ever officially added it.

**Comment by zero77** [comment]
Ok Thanks

**Comment by garfield69** [comment]
@zero77  has the multi-search-engine torrent dialog solved this ticket, or is this ticket's focus the auto-download service?

**Comment by zero77** [comment]
@garfield69 This ticket Is meant for the the auto-download service, not the multi-search-engine torrent dialog.

**Comment by garfield69** [comment]
*Definition of no-results*:     
Zero torrents. The AD monitor status shows 'nothing found'.     
Nothing found does not mean torrents that do not have enough seeders, or where the filters (Global include, exclude, quality, size) combine to prevent a torrent from being selected.           
*Requirements*:      
When enabled, if the default SE returns zero torrents, then try each of the other SE in turn, as specified in settings. The user can add/remove SE to be used for this search in the settings, but not alter the order they are used.    
- [ ] split the Auto-download options from the Settings-Torrent sidepanel into its own
- [ ] add option to enable using multiple SE when default SE returns zero torrents.
- [ ] add option to list all SE and allow use to cherckbox tick which SE are used in the search fall-back.
- [ ] amend auto-download service for multi SE use.
- [ ] provide translations for new setting options.

**Comment by zero77** [comment]
Not sure if this is what you mean by:

> amend auto-download service for multi SE use.

But can you disable the default SE and instead download the torrent with the highest seeds from any SE.

Also, I do know this is still in progress but I just spotted this part.

**Comment by garfield69** [comment]
My first attempt back in Oct 2016 for the auto-download multi-engine search turned into an unmitigated mess. Now that sufficient time has passed I will at some point take a fresh look and try again.      

>Not sure if this is what you mean by:
>>amend auto-download service for multi SE use.      

>But can you disable the default SE and instead download the torrent with the highest seeds from any SE.      

The goal here will be to allow the user to use settings to select one or more of the search engines that are defined to DuckieTV, for use by the auto-download when it performs a search for a torrent.     
The ADS will use all the selected SE for its episode torrent search and once the results are collected, the most suitable hit will be used for the download.
